homeassistant/components/switch/rainmachine.py

```diff
@@ -217,56 +194,62 @@ def name(self) -> str:
         """Return the name of the program."""
         return 'Program: {}'.format(self._entity_json.get('name'))
 
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return '{0}_program_{1}'.format(self.device_mac.replace(':', ''),
```

Comment:
Wouldn't just the `uid` be enough? (assuming that stands for "unique id". Also just the MAC would be enough.

Reply:
Would love your thoughts:

There are situations where one might have multiple sprinkler controllers (e.g., I have one and I manage one at another property, all through the same HASS interface). Let's say we have a program called "Landscaping" as the first program (UID: 1) for each controller – wouldn't there need to be some way to differentiate? I was thinking that the combination of the device MAC and the program/zone UID would cover everything. If that's overkill, let me know.

Reply:
Oh, I didn't realize that uid was unique _per instance_. Combining it is 👍